src/lib/components/BaseResponseElement/index.js

```diff
@@ -0,0 +1,256 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {BaseElement} from "../BaseElement";
+import "./_styles.scss";
+
+/**
+ * Base element used by all self-assessment response components.
+ *
+ * @extends {BaseElement}
+ */
+export class BaseResponseElement extends BaseElement {
```

Comment:
There's only one thing that extends this here. Do you expect there to be more?

Reply:
Yeah, I'm hoping there'll eventually be multiple response types (select input, drag-and-drop, etc.). The second type is coming in the next PR. So I thought it made sense to break out the validation logic and let each response type just worry about its rendering as much as is possible.